pkg/webui/console/views/device-general-settings/network-server-form/index.js

```diff
@@ -298,6 +356,16 @@ const NetworkServerForm = React.memo(props => {
         patch.mac_settings.use_adr = null
       }
 
+      if (patch.mac_settings.adr.dynamic?.overrides) {
```

Comment:
It's better to make use of `encode`/`decode` or `valueSetter` to get the form values into the desired shape. Keeping the form values as close as possible to the backend data structure will make the code less error-prone and also make validation more thorough. It also saves you from handling various edge cases to handle the fact that the form values are derived.
For these reasons, it's more idiomatic in how we do form logic (e.g. in the device onboarding form).